### ISSUE TITLE
`misc-const-correctness` not reported for pointer

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#58093]
The following reports nothing by default:

```cpp
#include <list>

class A
{
    public:
        void f() const;
};

extern void f(std::list<A> l)
{
    std::list<A>::iterator it;
    for(it = l.begin(); it != l.end(); ++it)
    {
        A* a = &(*it); // no warning
        a->f();
    }
}
```

This is also not reported if you completely remove the `f()` method.

If you enable `misc-const-correctness.WarnPointersAsValues` you get the warning on the pointer though:
```console
<source>:14:9: warning: variable 'a' of type 'A *' can be declared 'const' [misc-const-correctness]
        A* a = &(*it); // no warning
        ^
```

https://godbolt.org/z/KEbE6acvW

CC @JonasToth

**Comment by JonasToth** [comment]
Yes, this is a known linitation.
There is old unfinished work for that already in existence.

https://reviews.llvm.org/D52219

**Comment by firewave** [comment]
This is fixed in clang-tidy-21. Mostly likely by #130494.

 @HerrCai0907
